sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -253,6 +257,143 @@ case class FilterExec(condition: Expression, child: SparkPlan)
   }
 }
 
+/**
+ * Physical plan node for a recursive relation that encapsulates the physical plan of the anchor
+ * term and the logical plan of the recursive term.
+ *
+ * Anchor is used to initialize the query in the first run.
+ * Recursive term is used to extend the result with new rows, They are logical plans and contain
+ * references to the result of the previous iteration or to the so far accumulated result. These
+ * references are updated with new statistics and data and then compiled to physical plan before
+ * execution.
+ *
+ * The execution terminates once the anchor term or an iteration of the recursive term return no
+ * rows.
+ *
+ * @param cteName the name of the recursive relation
+ * @param anchorTerm this child is used for initializing the query
+ * @param output the attributes of the recursive relation
+ */
+case class RecursiveRelationExec(
+    cteName: String,
+    anchorTerm: SparkPlan,
+    output: Seq[Attribute]) extends SparkPlan {
+  @transient
+  lazy val logicalRecursiveTerm = logicalLink.get.asInstanceOf[RecursiveRelation].recursiveTerm
```

Comment:
We need to be aware that the logical link is not guaranteed to be present. It can be broken easily by physical rules in QueryExecution.preparations. It is true that AQE heavily relies on it, but it does sanity check beforehand and backs off if any logical link is missing.

Reply:
I see, thanks. Since I need the `logicalRecursiveTerm` to be there I can change this not to use the logical link.

Reply:
I think a `transient` member would do

Reply:
Ok, fixed.